src/main/kotlin/reactor/kotlin/extra/math/MathFluxExtensions.kt

```diff
@@ -101,10 +348,72 @@ fun <T> Flux<T>.sum(mapper: (T) -> Number): Mono<Long>
  *
  * @param mapper a lambda converting values to [Number]
  * @author Simon Baslé
- * @since 3.1.1
+ * @since 1.0.0
  */
 fun <T> Flux<T>.sumDouble(mapper: (T) -> Number): Mono<Double>
         = MathFlux.sumDouble(this, Function(mapper))
+
+/**
+ * Extension to map arbitrary values in a [Flux] to [Number]s and return the sum of these
+ * Numbers as a [Mono] of [BigInteger]
+ *
+ * Note that numbers will be mapped to [BigInteger] using Java standard conversions
+ * This may lead to arbitrary precision gain or loss if the source flux contains different types
+ * Please choose an appropriate method based on the expected types in the source flux.
+ *
+ * @param mapper a lambda converting values to [Number]
+ * @author Mark Pruden
+ * @since 1.1.5
+ */
+fun <T> Flux<T>.sumBigInt(mapper: (T) -> Number): Mono<BigInteger>
+        = MathFlux.sumBigInteger(this, Function(mapper))
+
+/**
+ * Extension to map arbitrary values in a [Flux] to [Number]s and return the sum of these
+ * Numbers as a [Mono] of [BigInteger]
+ *
+ * Note that numbers will be mapped to [BigInteger] using Java standard conversions
+ * This may lead to arbitrary precision gain or loss if the source flux contains different types
+ * Please choose an appropriate method based on the expected types in the source flux.
+ *
+ * @param mapper a lambda converting values to [Number]
+ * @author Mark Pruden
+ * @since 1.1.5
+ */
+fun <T> Flux<T>.sumBigDecimal(mapper: (T) -> Number): Mono<BigDecimal>
+        = MathFlux.sumBigDecimal(this, Function(mapper))
+
+/**
+ * General purpose extension function to map arbitrary values in a [Flux] to [Number]s and
+ * return the average of these Numbers as a [Mono] of [Number].
+ * The resultant [Mono] will have the same [Number] type as the output of the mapping function
+ *
+ * If the result type (or precision) needs to be changed then a specific averageType() method
+ * should be used in preference to this method. e.g. averageDouble(mapper) can be used to compute
+ * the average as a [Mono] of [Double]
+ *
+ * @param mapper a lambda converting values to [Number]
+ * @author Mark Pruden
+ * @since 1.1.5
+ */
+@Suppress("UNCHECKED_CAST")
+inline fun <T: Any, reified R: Number> Flux<T>.averageAll(noinline mapper: (T) -> R): Mono<R> =
+    when (R::class) {
+        BigDecimal::class -> MathFlux.averageBigDecimal(this, Function(mapper)) as Mono<R>
+        BigInteger::class -> MathFlux.averageBigInteger(this, Function(mapper)) as Mono<R>
+        Double::class -> MathFlux.averageDouble(this, Function(mapper)) as Mono<R>
+        Float::class -> MathFlux.averageFloat(this, Function(mapper)) as Mono<R>
+        Long::class -> MathFlux.averageBigInteger(this, Function(mapper))
```

Comment:
I assume this is for better precision of the _average_, an issue that doesn't occur in sums?

Reply:
Good pickup.

It is actually because the underlying Library does not support the averaging of integer type numbers, However for some reason BigInteger is supported for average.

Is this an omission - sort of IMHO - While most people think of averages and decimal numbers (obvious reasons), an average expressed as an integer is still an average (just with less precision) 

For completeness am using Big integer to do the averages then map the result back to the requested type.